semgrep-core/bin/Main.ml

```diff
@@ -530,7 +549,16 @@ let iter_generic_ast_of_files_and_get_matches_and_exn_to_errors f files =
          let ast = parse_generic lang file in
 
          (* calling the hook *)
-         f file lang ast, []
+         run_with_memory_limit !maxout (fun () ->
+           (f file lang ast, [])
+           (* to test -maxout, you can uncomment the code below, to give
+            * a chance to the Gc.create_alarm to run even if the program does
+            * noi even need to run the Gc. However, this has a slow perf
+            * penality on small programs, which is why it's better to keep
+            * it in comment when you're not testing -maxout.
+            *)
+           (*|> (fun v -> Gc.full_major(); v) *)
+         )
```

Comment:
It would be nice to have a permanent test for this. You could add a `--test` or `--debug` flag that would enable the `Gc.full_major ()` (or an environment variable).